From: TDVP <tdvp@example.invalid>
Date: Wed, 26 Aug 2026 16:45:00 +0800
Subject: [PATCH] tdvp: build only the wf-panel-pi power module

Keep Raspberry Pi's upstream power-warning implementation while omitting its
unused LXPanel target and dependency.

Signed-off-by: TDVP <tdvp@example.invalid>
---
 src/meson.build | 19 +++----------------
 1 file changed, 3 insertions(+), 16 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index 50740ec..93a180e 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -1,26 +1,9 @@
-gtk = dependency('gtk+-3.0')
 gtkmm = dependency('gtkmm-3.0', version: '>=3.24')
-lxpanel = dependency('lxpanel-pi')
 wfpanel = dependency('wf-panel-pi')
 udev = dependency('libudev')
 
-lsources = files(
-  'power.c'
-)
-
-ldeps = [ gtk, lxpanel, udev ]
-
-largs = [ '-DPACKAGE_DATA_DIR="' + lresource_dir + '"', '-DGETTEXT_PACKAGE="lpplug_' + meson.project_name() + '"' ]
-
-shared_module(meson.project_name(), lsources,
-        dependencies: ldeps,
-        install: true,
-        install_dir: get_option('libdir') / 'lxpanel-pi',
-        c_args : largs,
-        name_prefix: ''
-)
-
-wsources = files(
+sources = files(
+  'power.c',
   'power.cpp'
 )
 
@@ -28,7 +11,7 @@ wdeps = [ gtkmm, wfpanel , udev ]
 
 wargs = [ '-DPACKAGE_DATA_DIR="' + wresource_dir + '"', '-DGETTEXT_PACKAGE="wfplug_' + meson.project_name() +'"' ]
 
-shared_module('lib' + meson.project_name(), [ lsources, wsources ],
+shared_module('lib' + meson.project_name(), sources,
         dependencies: wdeps,
         install: true,
         install_dir: get_option('libdir') / 'wf-panel-pi',
